django/db/backends/postgresql/operations.py

```diff
@@ -282,3 +282,14 @@ def explain_query_prefix(self, format=None, **options):
 
     def ignore_conflicts_suffix_sql(self, ignore_conflicts=None):
         return 'ON CONFLICT DO NOTHING' if ignore_conflicts else super().ignore_conflicts_suffix_sql(ignore_conflicts)
+
+    def for_update_sql(self, nowait=False, skip_locked=False, of=(), no_key=False):
+        """
+        Return the FOR UPDATE SQL clause to lock rows for an update operation.
+        """
+        return 'FOR%s UPDATE%s%s%s' % (
+            ' NO KEY' if no_key else '',
+            ' OF %s' % ', '.join(of) if of else '',
+            ' NOWAIT' if nowait else '',
+            ' SKIP LOCKED' if skip_locked else '',
+        )
```

Comment:
I would handle `NO KEY` in `BaseDatabaseOperations.or_update_sql()` instead of repeating everything here.